### PR TITLE
Upgrade React, Remix, remix-auth-oauth2, fix scope bug

### PR DESCRIPTION
**Pull request by Hansenq**
Fixes https://github.com/pbteja1998/remix-auth-google/issues/14, upgrades to latest version of `remix-auth-oauth2`

Upgrades React to v18 and Remix to v2 as well to fix dependency issues.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -120,7 +118,6 @@ export class GoogleStrategy<User> extends OAuth2Strategy<
 
   protected authorizationParams(): URLSearchParams {
     const params = new URLSearchParams({
-      scope: this.scope.join(GoogleStrategyScopeSeperator),
```
Comment (Hansenq) [inline src/index.ts:123]: this is now handled in `remix-auth-oauth2`
https://github.com/sergiodxa/remix-auth-oauth2/blob/8f5e6a44835128c088435f9a7da2c60dc7dbcf84/src/index.ts#L368

**Comment by pbteja1998** [comment]
@Hansenq have you tested all the changes made in the PR? 

Also, please don’t do the version bump in this PR. We will be doing it separately.

**Comment by Hansenq** [comment]
Yup, just tested it with my remix app installing it from my repo and it works. There aren't any unit tests to run. Reverted version bump.

**Comment by pbteja1998** [comment]
@Hansenq Can you please remove `package-lock.json` and run `yarn install` once. That will update `yarn.lock` file which is what we are using in this repo.

**Comment by Hansenq** [comment]
ah done.

**Comment by pbteja1998** [comment]
Thank you @Hansenq 🙏

Published this in `v1.3.0`

**Comment by MichaelDeBoey** [comment]
@pbteja1998 This is a **breaking** change.
Please revert & publish a new minor version

Once that's done, you can revert the revert & publish a major version to be semver compliant again

**Comment by pbteja1998** [comment]
@MichaelDeBoey is https://github.com/pbteja1998/remix-auth-email-link/pull/42 also a breaking change? 

Should I have to do the same thing there too? Or is it fine?

**Pull request merged**